### PR TITLE
[Enhancement] Support no-IRQ for SPI Ethernet W5500

### PR DESCRIPTION
**Pull request by TD-er**
Requires this commit from ESP-IDF to be cherry picked: https://github.com/espressif/esp-idf/commit/fd0a1dc53c4cff99f04dfe658047f7c2e3e6d8ad

Related discussion: https://github.com/espressif/esp-idf/issues/12682#issuecomment-1914807052

Fixes: https://github.com/espressif/arduino-esp32/issues/9246

### CODE REVIEW

**Comment by Jason2866** [comment]
@TD-er  This is not in official version (Tasmota fork only)
```
case ETH_PHY_JL1101:
            phy = esp_eth_phy_new_jl1101(&phy_config);
            break;
```
Needs to be removed.

**Comment by me-no-dev** [comment]
this will have to first be backported from IDF.

File: libraries/Ethernet/src/ETH.cpp [hunk 1]
```diff
@@ -351,7 +351,7 @@ bool ETHClass::beginSPI(eth_phy_type_t type, uint8_t phy_addr, int cs, int irq,
         log_w("ETH Already Started");
         return true;
     }
-    if(cs < 0 || irq < 0){
+    if(cs < 0 /*|| irq < 0*/){
         log_e("CS and IRQ pins must be defined!");
```
Comment (me-no-dev) [inline libraries/Ethernet/src/ETH.cpp:355]: remove the commented check and fix the error message

**Comment by TD-er** [comment]
> this will have to first be backported from IDF.

Absolutely, but when that's done it is very easy to implement :)

**Comment by me-no-dev** [comment]
@TD-er updated the PR to be able to merge it now and enable the feature with define, once it's available from IDF

**Comment by TD-er** [comment]
Ah good to know, so I should not forget to add those defines in my own builds or else it won't work with my own builds when updating the Arduino code :)

**Comment by me-no-dev** [comment]
@TD-er you can add `#define ETH_SPI_SUPPORTS_NO_IRQ 1` to `ETH.h` or globally define it in the build to enable the feature

**Comment by TD-er** [comment]
@TD-er Yep, already add it to my platformIO config or else I'm sure I will forget it later :)
Thanks for merging.

**Pull request merged**